test/test_sparse.py

```diff
@@ -3364,21 +3364,21 @@ class TestSparseOneOff(TestCase):
     def test_cuda_from_cpu(self):
         with self.assertRaisesRegex(
                 RuntimeError,
-                "backend of indices \\(CUDA\\) must match backend of values \\(CPU\\)"):
+                "Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"):
             torch.sparse.FloatTensor(torch.zeros(1, 4).long().cuda(),
                                      torch.randn(4, 4, 4),
                                      [3, 4, 4])
 
         with self.assertRaisesRegex(
                 RuntimeError,
-                "backend of indices \\(CUDA\\) must match backend of values \\(CPU\\)"):
+                "Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"):
             torch.sparse.FloatTensor(torch.zeros(1, 4).long().cuda(),
                                      torch.randn(4, 4, 4, 0),
                                      [3, 4, 4, 0])
 
         with self.assertRaisesRegex(
                 RuntimeError,
-                "backend of indices \\(CUDA\\) must match backend of values \\(CPU\\)"):
+                "Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"):
```

Comment:
These error message changes are because of the way `CUDA` and `SparseCPU` interact. We effectively end up `or`ing together `DispatchKey::CUDA` and `DispatchKey::SparseCPU` together - which in the old scheme creates a new keyset with `SparseCPU` as the highest priority, but in the new scheme the highest priority key is now `SparseCUDA`.

The error message changes because we dispatch o the `SparseCUDA` kernel, which has codegen'd device checks (which get hit here, with the new error message). Previously, we'd hit the `SparseCPU` kernel, which had no device checks, then eventually fail later when we try to construct the actual `SparseTensorImpl` object.